src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentReferenceCaptureLoweringPass.cs

```diff
@@ -33,32 +35,106 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
+                    // already be a diagnostic for empty values
+                    var identifier = DetermineIdentifierToken(node);
+                    if (identifier == null)
+                    {
+                        continue;
+                    }
+
+                    var rewritten = RewriteUsage(reference.Parent, identifier);
+                    reference.Replace(rewritten);
+
+                    // Now we need to check if the field generation has been suppressed.
+                    //
+                    // You have to suppress field generation for generic types because we don't know the
+                    // type name to create the field.
+                    var generateField = ShouldGenerateField(reference.Parent);
+
+                    // Insert the field with other fields, near the top of the class.
+                    if (generateField)
+                    {
+                        var position = 0;
+                        while (position < @class.Children.Count && @class.Children[i] is FieldDeclarationIntermediateNode)
+                        {
+                            i++;
+                        }
+
+                        @class.Children.Insert(position, CreateField(rewritten.FieldTypeName, identifier));
+                    }
                 }
             }
         }
 
-        private IntermediateNode RewriteUsage(ClassDeclarationIntermediateNode classNode, IntermediateNode parent, TagHelperDirectiveAttributeIntermediateNode node)
+        private ReferenceCaptureIntermediateNode RewriteUsage(IntermediateNode parent, IntermediateToken identifier)
         {
-            // If we can't get a nonempty attribute name, do nothing because there will
-            // already be a diagnostic for empty values
-            var identifierToken = DetermineIdentifierToken(node);
-            if (identifierToken == null)
-            {
-                return node;
-            }
-
             // Determine whether this is an element capture or a component capture, and
             // if applicable the type name that will appear in the resulting capture code
             var componentTagHelper = (parent as ComponentIntermediateNode)?.Component;
             if (componentTagHelper != null)
             {
-                return new ReferenceCaptureIntermediateNode(identifierToken, componentTagHelper.GetTypeName());
+                return new ReferenceCaptureIntermediateNode(identifier, componentTagHelper.GetTypeName());
             }
             else
             {
-                return new ReferenceCaptureIntermediateNode(identifierToken);
+                return new ReferenceCaptureIntermediateNode(identifier);
+            }
+        }
+
+        private bool ShouldGenerateField(IntermediateNode parent)
+        {
+            var parameters = parent.FindDescendantNodes<TagHelperDirectiveAttributeParameterIntermediateNode>();
+            for (var j = 0; j < parameters.Count; j++)
```

Comment:
Team `j`!

Reply:
Oh look. it's **A** J. 😆 

Will change this to be something more.... expected.